ompi/mca/io/romio314/src/io_romio314_component.c

```diff
@@ -234,15 +235,29 @@ static int delete_query(const char *filename, struct ompi_info_t *info,
 }
 
 
-static int delete_select(const char *filename, struct ompi_info_t *info,
+static int delete_select(const char *filename, struct opal_info_t *info,
                          struct mca_io_base_delete_t *private_data)
 {
     int ret;
 
+// Technically an opal_info_t isn't a full ompi_info_t. so if we're
+// using an MPI call below with an MPI_Info, we need to create an equivalent
+// MPI_Info:
+// In the current design a lot of code is written to use opal_info_t.
+// I'd be open to reverting most of it back to ompi_info_t and using
+// opal_info_* calls on the opal_info_t that's inside the ompi_info_t.
+// I don't think there's any advantage to most of the code being switched
+// to opal_info_t, which is causing the unnecessary extra complexity here:
```

Comment:
I guess a decision needs made for what is the predominant design. Dave has ompi_info_t as the larger class that contains an opal_info_t plus a tiny bit extra (like a fortran int id). All the top level MPI calls that use info come in with an ompi_info_t*.

Dave's overall design is to migrate toward the code from that point down being largely opal_info_t. I don't really disagree, although it does end up with an awful lot of
  ```some_function(mpiinfo)```  becoming  ```some_function(&mpiinfo->super)``` // to get the opal_info_t
and equivalently, if you just want to pass MPI_INFO_NULL to some routine, but it wants an opal_info_t*, you have to give it &MPI_INFO_NULL->super.

And there are a tiny number of places where the lower level code calls back into MPI, and thus needs the larger ompi_info_t when all it has is an opal_info_t, thus having the extra code you quoted that's making another copy of the info that isn't logically necessary. That bothers me, but it's only 3 places... so I guess I'm leaning toward reverting my comment, and sticking with the design of having the lower level code predominantly using opal_info_t.